lib/quadstore.ts

```diff
@@ -249,45 +249,44 @@ export class Quadstore implements Store {
         return indexBatch.put(index.getKey(importedQuad), value);
       }, quadBatch);
     }, this.db.batch());
-    await pFromCallback((cb) => { batch.write(cb); });
+    await this.writeBatch(batch, opts);
     return { type: ResultType.VOID };
   }
 
-  async del(quad: Quad, opts: EmptyOpts = emptyObject): Promise<VoidResult> {
+  async del(quad: Quad, opts: BatchOpts = emptyObject): Promise<VoidResult> {
     this.ensureReady();
     const batch = this.indexes.reduce((batch, i) => {
       return batch.del(i.getKey(importQuad(quad, this.defaultGraph, this.prefixes)));
     }, this.db.batch());
-    await pFromCallback((cb) => { batch.write(cb); });
+    await this.writeBatch(batch, opts);
     return { type: ResultType.VOID };
   }
 
-  async multiDel(quads: Quad[], opts: EmptyOpts = emptyObject): Promise<VoidResult> {
+  async multiDel(quads: Quad[], opts: BatchOpts = emptyObject): Promise<VoidResult> {
```

Comment:
Instead of using `BatchOpts` as a single type for all `opts` objects, could you extend that into `DelOpts`, `PutOpts` and `PatchOpts` and use the more specialized ones in the respective methods? It doesn't make a difference right now but it paves the way to further work on `opts` params.